Patch command not found despite being found.

Hello,

the current master e5d28807ef5a94987bc0c0cf60b3f16bb60adce1 claims that there is no suitable `patch` command available, despite locating two patch applications with suitable versions right before that statement.  Looking into the code I don't see any reason why it shouldn't work.

```
-- Looking for CMake property POSITION_INDEPENDENT_CODE - found
-- SRC_DIR: /Users/travis/build/[secure]/[secure]-sdk/build/python-prefix/src/python
-- PY_VERSION: 2.7.14
-- Found Git: /Applications/Xcode.app/Contents/Developer/usr/bin/git (found version "2.3.2 (Apple Git-55)") 
-- Found Patch: /usr/local/bin/patch (found version "2.7.6") 
CMake Error at cmake/PythonApplyPatches.cmake:41 (message):
  Could NOT find a suitable version of Git or Patch executable to apply
  patches.
Call Stack (most recent call first):
  CMakeLists.txt:293 (include)
-- Configuring incomplete, errors occurred
```

The full log can be [found here with the relevant chunk highlighted](https://travis-ci.org/freeorion/freeorion-sdk/jobs/357719598#L6382-L6416).

--- a/cmake/PythonApplyPatches.cmake
+++ b/cmake/PythonApplyPatches.cmake
@@ -6,7 +6,7 @@ set(CMAKE_MODULE_PATH
 
 if(NOT DEFINED PATCH_COMMAND)
   find_package(Git)
-  if(Git_FOUND)
+  if(Git_FOUND OR GIT_FOUND)
     set(PATCH_COMMAND ${GIT_EXECUTABLE} apply --whitespace=fix)
     # Initialize Git repo to ensure "git apply" works when source tree
     # is located within an already versioned tree.
@@ -25,7 +25,7 @@ if(NOT DEFINED PATCH_COMMAND)
     endif()
   else()
     find_package(Patch)
-    if(Patch_FOUND)
+    if(Patch_FOUND OR PATCH_FOUND)
       # Since support for git diffs which copy or rename files was
       # added in patch 2.7, we can not use older version.
       if("${Patch_VERSION}" VERSION_EQUAL "2.7.0" OR "${Patch_VERSION}" VERSION_GREATER "2.7.0")

